Sources/PubNubChatComponents/Previews/SharedPreviewData.swift

```diff
@@ -0,0 +1,186 @@
+//
+//  File 2.swift
+//  
+//
+//  Created by Jakub Guz on 9/19/22.
+//
+
+#if canImport(SwiftUI) && DEBUG
+
+import Foundation
+import PubNub
+import PubNubChat
+import CoreData
+import Combine
+
+// The idea could be creating a shared class (or something similar) that holds shared data.
+// Then it's up to you what data you would like to use in your previews. Please don't treat existing implementation very serious, it's rather a POC 
+
+// MARK: ChatProvider with in-memory storage
+
+public let testChatProvider: PubNubChatProvider = {
+  
+  let chatProvider = PubNubChatProvider(
+    pubnubProvider: PubNub(configuration: PubNubConfiguration(publishKey: "...", subscribeKey: "...", userId: "JG")),
+    coreDataProvider: try! CoreDataProvider(location: .memory, flushDataOnLoad: false),
+    cacheProvider: UserDefaults.standard
+  )
+  
+  let channel = PubNubChatChannel(
+    id: "CHANNEL1",
+    name: "Channel 1",
+    type: "direct",
+    status: "status",
+    details: "Channel details",
+    avatarURL: URL(string: "https://picsum/photos/200/300"),
+    updated: nil,
+    eTag: nil,
+    custom: VoidCustomData()
+  )
+  
+  let user1 = PubNubChatUser(
+    id: "JG",
+    name: "JG"
+  )
+  
+  let message = PubNubChatMessage(
+    id: "12345",
+    text: "Hello, world!!!",
+    pubnubUserId: "JG",
+    pubnubChannelId: "CHANNEL1"
+  )
+  
+  let membership = PubNubChatMember(
+    channelId: "CHANNEL1",
+    userId: "JG"
+  )
+  
+  chatProvider.dataProvider.load(users: [user1])
+  chatProvider.dataProvider.load(members: [membership])
+  chatProvider.dataProvider.load(channels: [channel])
+  chatProvider.dataProvider.load(messages: [message], processMessageActions: false)
+  
+  return chatProvider
+  
+}()
+
+// MARK: ManagedMessageViewModel conformance
+
+class MyViewModel: ManagedMessageViewModel {
+  
+  public typealias Entity = PubNubManagedMessage
+  public typealias ChannelViewModel = PubNubManagedChannel
+  public typealias UserViewModel = PubNubManagedUser
+  public typealias MessageActionModel = PubNubManagedMessageAction
+  
+  public var pubnubId: Timetoken { return 0 }
+  public var managedObjectId: NSManagedObjectID { return NSManagedObjectID() }
+  
+  public func decodedContent<T: Decodable>(from: T.Type) throws -> T {
+    return try JSONDecoder().decode(T.self, from: Data())
```

Comment:
we need decision if we using return in single lines or not

Reply:
We have been historically including the return, but we can default to the standard when we add SwiftLint and/or SwiftFormat plugin support.